### DISCUSSION TITLE
north eastern us zip codes too short

### DISCUSSION THREAD

**Discussion by ahaberbosch** [faker-js/faker#2276] [Q&A] [answered]
I realized in my testing that some north eastern us zip codes only return 4 numbers, because they officially start with zero. This means they fail most forms. Is there a way to correct this with a custom locale pattern or similar rather than an after-the-fact padStart?

**Comment by matthewmayer** [comment]
Do you have sample code? Are you using the en or en_US locale?

**Reply by ahaberbosch** [reply]
`faker.location.zipCode({state: 'NH'})` 

a quick google search suggests these are the other jurisdictions that may have similar zipcodes:

> Connecticut, Massachusetts, Maine, New Hampshire, New Jersey, Puerto Rico, Rhode Island, Vermont, Virgin Islands, Army Post Office Europe, Fleet Post Office Europe

I am using en_US locale

**Reply by matthewmayer** [reply, marked as answer]
Added as an issue at https://github.com/faker-js/faker/issues/2277
